src/main/java/com/meta/cp4m/S3PreProcessor.java

```diff
@@ -75,11 +81,11 @@ public ThreadState<T> run(ThreadState<T> in) {
 
   public void sendRequest(byte[] media, String senderID, String extension) {
     String key = senderID + '_' + Instant.now().toEpochMilli() + '.' + extension;
-
+    AwsCredentialsProvider credentialsProvider = Objects.requireNonNullElse(this.staticCredentialsProvider, DefaultCredentialsProvider.create());
```

Comment:
why not push this logic up into the constructor so that you don't have to do a null check here?

Reply:
Moved this to constructor